### ISSUE TITLE
Observe for data change

### ISSUE THREAD

**Issue by vihai** [miguelcobain/ember-yeti-table#3]
Hello,

I wrote an object streaming layer that keeps the models in the store updated. Yeti-table loads the models from the store or via the adapter via loadData, eventually applying pagination and such.

When models are updated everything works fine. A different story is when an object is added or destroyed.

There should be some way to inform the table component that the collection has changed in composition to trigger a new loadData that will update contents and totalRows.

How would you approach this issue?

Thanks!

**Comment by miguelcobain** [comment]
Hmm. If the models are already in the store, why not pass them to the `@data` argument?

Using something like
```
data: computed(function() {
  return this.store.peekAll('model-name');
})
```

and passing it to yeti table. As new records arrive, I think the array returned by peekAll is updated with them, and consequently, yeti-table also updates.

**Comment by vihai** [comment]
Sorry for the delay. I do not pass the models via the @data argument because I need both pagination and the ability to load and keep in the store only the shown models.

Ideally I would pass and object similar to a RecordArray and dynamically handle retrieval and unload from the store, do you think it would be viable and better approach than using loadData ?

**Comment by miguelcobain** [comment]
> I do not pass the models via the @data argument because I need both pagination and the ability to load and keep in the store only the shown models.

It seems like you want the `@loadData` argument indeed, yes.

I'm not exactly understanding your use case. Maybe you could show me some code that illustrates what you're trying to do and what the problem is?

**Comment by vihai** [comment]
My use case can be described like this:

* I wrote an object streaming layer that implements a binding concept between the objects on server and the objects in the local store. Updated objects are pushed to the client automatically, destroyed objects are unloaded from the store. This is pretty much transparent thanks to computed properties and other Ember tricks.
* I have collections containing thousands or tens of thousands of models.
* I cannot bind to the whole collection due to its big size, I thus only get and bind to a selection of models specified with filters and/or pagination.
* Sometimes I receive notifications of new models matching the filtering criteria.

I would like to show this many objects in a grid/table component I am porting some code that comes from ExtJS that has a very powerful table with infinite scrolling.

I think a more flexible approach would be to drop `@loadData` and assume that `@data` can an array or an object a bit smarter than an array with some property that can be observed to react to changes in the the collection itself.

I will also experiment with ember-infinity to see if that approach could be viable.

Many grid/table components for ember are either badly implemented, half done, abandoned. Yours, OTOH seems well designed thus I'm investing all my hopes in it!

**Comment by hbrysiewicz** [comment]
@vihai I've got what I think is the same/similar issue in #197 

I was able to work-around it for now. 

Here is my scenario in case it helps you with yours (or anyone landing here in the future):

I have row actions inside the table that will add/remove/mutate rows. This action won't trigger the `loadData` method on the table, so I need some way to "force" a reload of the data. I found that `table.rows.query` was what I needed when inside the context of the table.  Since `rows` is the model array proxy, I can take the `query` and use this on the action to refresh from the server.
